File: src/functions/fun3-git-node-public/handler.js
const fibonacci = require ('fibonacci');

module.exports = {
    main: async function (event, context) {
        const dataIn = sanitizeInput(event.data);
        var payload = {
            ...dataIn,
            check_in: {
                ...dataIn.check_in,
                fun3: new Date()
            },
            fibo: {
            ...dataIn.fibo,
            fun3: fibonacci.iterate(between(parseInt(process.env['FIBONACCI_MIN'],10),parseInt(process.env['FIBONACCI_MAX'],10)))
            }
        };

        var eventType = process.env['PUSH_EVENT_TYPE'];
        var eventSource = event.extensions.request.get('ce-source')
        event.emitCloudEvent(eventType, eventSource, payload);
        // console.log(`Payload [${payload.uuid}] pushed to ${process.env['PUSH_EVENT_TYPE']}`,payload)
        return "OK";
    },
  };


  function between(min, max) {
    return Math.floor(Math.random() * (max - min) + min);
  }
  
  function sanitizeInput(input) {
    console.log(input)

    let sanitised = input;
    console.log("Type of incomming input : "+ typeof sanitised);
    return sanitised;
}

File: src/functions/fun2-inline-python/handler.py
import os
import datetime
import json

def main(event, context):
    payload=event['data']
    if isinstance(payload, (bytes, bytearray)):
        payload=json.loads(str(payload, "utf-8"))
    checkin=payload['check_in']
    fibo=payload['fibo']
    checkin['fun2']=datetime.datetime.now().isoformat()
    fibo['fun2']=recur_fibo(10)
    eventSource = "kyma"
    eventType = os.environ['PUSH_EVENT_TYPE']
    event.emitCloudEvent(eventType, eventSource, payload)
    print(payload)
    return "OK"


def recur_fibo(n):
   if n <= 1:
       return n
   else:
       return(recur_fibo(n-1) + recur_fibo(n-2))